test/e2e/storage/pod_add_delete_add_fast.go

```diff
@@ -0,0 +1,177 @@
+		ns                string
+		pvc               *v1.PersistentVolumeClaim
+		nodeName          string
+		isNodeLabeled     bool
+		nodeKeyValueLabel map[string]string
+		nodeLabelValue    string
+	)
+
+	f := framework.NewDefaultFramework("pv")
+
+	// We are going to pause between pod creation and deletion for this duration because
+	// while we do want attach/detach controller to initiate action based on events
+	// we do not want pod to be fully running or fully deleted because in either will
+	// cause volume to be attached or detached.
+	pauseBetweenAction := 10 * time.Second
+
+	BeforeEach(func() {
+		c = f.ClientSet
+		ns = f.Namespace.Name
+		framework.SkipUnlessProviderIs("gce", "gke", "aws")
+
+		nodeList := framework.GetReadySchedulableNodesOrDie(f.ClientSet)
+		if len(nodeList.Items) != 0 {
+			nodeName = nodeList.Items[0].Name
+		} else {
+			framework.Failf("Unable to find ready and schedulable Node")
+		}
+
+		if !isNodeLabeled {
+			nodeLabelValue := "pod_vol_add_del_add_" + string(uuid.NewUUID())
+			nodeKeyValueLabel = make(map[string]string)
+			nodeKeyValueLabel["pod_vol_add_del_add_label"] = nodeLabelValue
+			framework.AddOrUpdateLabelOnNode(c, nodeName, "pod_vol_add_del_add_label", nodeLabelValue)
+			isNodeLabeled = true
+		}
+
+		// Make sure there is a default storageclass
+		defaultScName := getDefaultStorageClassName(c)
+		verifyDefaultStorageClass(c, defaultScName, true)
+		test := storageClassTest{
+			name:      "default",
+			claimSize: "2Gi",
+		}
+
+		pvc = newClaim(test, ns, "default")
+		updatedPVC, err := c.CoreV1().PersistentVolumeClaims(pvc.Namespace).Create(pvc)
+		pvc = updatedPVC
+		Expect(err).NotTo(HaveOccurred())
+		Expect(pvc).ToNot(Equal(nil))
+	})
+
+	framework.AddCleanupAction(func() {
+		if len(nodeLabelValue) > 0 {
+			framework.RemoveLabelOffNode(c, nodeName, "pod_vol_add_del_add_label")
+		}
+	})
+
+	AfterEach(func() {
+		framework.Logf("AfterEach: Cleaning up resources for fast add delete")
+
+		if c != nil {
+			if errs := framework.PVPVCCleanup(c, ns, nil, pvc); len(errs) > 0 {
+				framework.Failf("AfterEach: Failed to delete PVC and/or PV. Errors: %v", utilerrors.NewAggregate(errs))
+			}
+			pvc, nodeName, isNodeLabeled, nodeLabelValue = nil, "", false, ""
+			nodeKeyValueLabel = make(map[string]string)
+		}
+
+	})
+
+	It("should test that quickly adding, deleting and then adding back a pod does not stop pod creation", func() {
+		podCommand := "while true; do date; date >>/mnt/volume1; sleep 1; done"
+		pod1 := createPodWithNodeAffinity(pvc.Name, nodeKeyValueLabel, podCommand)
+		pod1, err := c.CoreV1().Pods(ns).Create(pod1)
+
+		Expect(err).NotTo(HaveOccurred())
+
+		time.Sleep(pauseBetweenAction)
+
+		// now lets delete the pod
+		podDeleteErr := c.CoreV1().Pods(ns).Delete(pod1.Name, nil)
+		Expect(podDeleteErr).NotTo(HaveOccurred())
+
+		time.Sleep(pauseBetweenAction)
+
+		pod2 := createPodWithNodeAffinity(pvc.Name, nodeKeyValueLabel, podCommand)
```

Comment:
why want to place the pod in the same node?

Reply:
The reason I want to place the pod in same node is because - I want to trigger the potential race where new pod (created in last step) starts to mount a volume because it sees volume in node's attached state and at the same time ADC detaches the volume thinking nobody is using it.

We know that ^ race currently doesn't exist, but idea is to try and hope to reproduce it.